agent/stats/engine_integ_test.go

```diff
@@ -220,89 +191,275 @@ func TestStatsEngineWithNewContainers(t *testing.T) {
+	containers := []*api.Container{
+		{
+			Name:            "container-health",
+			HealthCheckType: "docker",
+		},
+	}
+	testTask := &api.Task{
+		Arn:                 "container-health-task",
+		DesiredStatusUnsafe: api.TaskRunning,
+		KnownStatusUnsafe:   api.TaskRunning,
+		Family:              "docker-container-health",
+		Version:             "1",
+		Containers:          containers,
 	}
+	// Populate Tasks and Container map in the engine.
+	dockerTaskEngine, _ := taskEngine.(*ecsengine.DockerTaskEngine)
+	dockerTaskEngine.State().AddTask(testTask)
+	dockerTaskEngine.State().AddContainer(
+		&api.DockerContainer{
+			DockerID:   container.ID,
+			DockerName: "container-health",
+			Container:  containers[0],
+		},
+		testTask)
+
+	// Simulate container start prior to listener initialization.
+	time.Sleep(checkPointSleep)
+	err = engine.MustInit(taskEngine, defaultCluster, defaultContainerInstance)
+	assert.NoError(t, err, "initializing stats engine failed")
+	defer engine.containerChangeEventStream.Unsubscribe(containerChangeHandler)
+
+	// Wait for the stats collection go routine to start.
+	time.Sleep(checkPointSleep)
+
+	// Verify the metrics of the container
+	metadata, taskMetrics, err := engine.GetInstanceMetrics()
+	assert.NoError(t, err, "error gettting instance metrics")
+	assert.NoError(t, validateMetricsMetadata(metadata), "error validating metadata")
+	assert.Len(t, taskMetrics, 1, "incorrect number of tasks")
+
 	taskMetric := taskMetrics[0]
-	if *taskMetric.TaskDefinitionFamily != taskDefinitionFamily {
-		t.Errorf("Excpected task definition family to be: %s, got: %s", taskDefinitionFamily, *taskMetric.TaskDefinitionFamily)
+	assert.Equal(t, aws.StringValue(taskMetric.TaskDefinitionFamily), "docker-container-health", "task definition family not expected")
+	assert.Equal(t, aws.StringValue(taskMetric.TaskDefinitionVersion), taskDefinitionVersion, "task definition family version not expected")
+	assert.NoError(t, validateContainerMetrics(taskMetric.ContainerMetrics, 1), "error validating container metrics")
+
+	// Verify the health metrics of container
+	healthMetadata, healthMetrics, err := engine.GetTaskHealthMetrics()
+	assert.NoError(t, err, "getting task health metrics failed")
+	require.Len(t, healthMetrics, 1)
+	assert.NoError(t, validateHealthMetricsMetadata(healthMetadata))
+	assert.Equal(t, aws.StringValue(healthMetrics[0].TaskArn), testTask.Arn, "task arn not expected")
+	assert.Equal(t, aws.StringValue(healthMetrics[0].TaskDefinitionFamily), "docker-container-health", "task definition family not expected")
+	assert.Equal(t, aws.StringValue(healthMetrics[0].TaskDefinitionVersion), taskDefinitionVersion, "task definition version not expected")
+	assert.NoError(t, validateTaskHealth(healthMetrics[0].Containers, 1))
+
+	err = client.StopContainer(container.ID, defaultDockerTimeoutSeconds)
+	assert.NoError(t, err, "stopping container failed")
```

Comment:
I think all errors from docker client should instead be handled by `require` than `assert`.

Reply:
@aaithal, so i understand reasoning: why require vs assert?